### PR TITLE
Add a callback for gcs table lookup failures.

### PR DESCRIPTION
**Pull request by elibol**
Adds a callback for GCS table lookup failures.

### CODE REVIEW

**Review by pcmoritz — APPROVED** [review]

**Comment by pcmoritz** [comment]
Can be merged when the tests pass (the last one is failing at the moment and needs to be fixed).

File: src/ray/gcs/redis_context.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ namespace gcs {
 
 class RedisCallbackManager {
  public:
-  using RedisCallback = std::function<void(const std::string &)>;
+  using RedisCallback = std::function<void(const std::string &, bool is_nil)>;
```
Comment (stephanie-wang) [inline src/ray/gcs/redis_context.h]: Does it make sense to add this parameter for all `RedisCallbacks`? I'm a little wary of adding this when it is only used by one of the table operations.

An alternative plan would be to call the failure callback if the response is empty in the `Lookup` call. Then we can keep the changes completely local to the `Lookup` operation. The only drawback I see here is that if we won't be able to support an entry that has an empty string as the value, but I think this is probably okay.

**Review by stephanie-wang — CHANGES_REQUESTED** [review]

**Review by stephanie-wang — APPROVED** [review]
Looks good!

**Comment by elibol** [comment]
Tests on private travis passed.

**Pull request merged**